### PR TITLE
feat: add sign fn for sign_message

### PR DESCRIPTION
**Pull request by ChrisCho-H**
While it's not hard to create the signature using `secp256k1` modules with `signed_msg_hash`, it's much more convenient and safe to provide one-way function to generate signed signature(even without the understanding about the semantics of bitcoin message signing).

### CODE REVIEW

**Review by apoelstra — APPROVED** [review]
ACK 99d8476aba5db7e354b374ebe52d2c25cf7db347 successfully ran local tests; sure

File: bitcoin/src/sign_message.rs [hunk 1]
```diff
@@ -214,6 +215,16 @@ pub fn signed_msg_hash(msg: impl AsRef<[u8]>) -> sha256d::Hash {
     sha256d::Hash::from_engine(engine)
 }
 
+/// Sign message using Bitcoin's message signing format.
+#[cfg(feature = "secp-recovery")]
+pub fn sign(msg: impl AsRef<[u8]>, privkey: SecretKey) -> MessageSignature {
+    let secp = secp256k1::Secp256k1::new();
```
Comment (tcharding) [inline bitcoin/src/sign_message.rs:221]: Shouldn't this be a parameter?

**Comment by ChrisCho-H** [comment]
add secp_ctx as param!

**Review by apoelstra — APPROVED** [review]
ACK c41a6e9b1bdc2f60b80519ab5b3fbb9f0eb46acb successfully ran local tests

**Review by tcharding — APPROVED** [review]
ACK c41a6e9b1bdc2f60b80519ab5b3fbb9f0eb46acb

**Pull request merged**